Library/Formula/purescript.rb

```diff
@@ -0,0 +1,34 @@
+require "language/haskell"
+
+class Purescript < Formula
+  include Language::Haskell::Cabal
+
+  homepage "http://www.purescript.org"
+  url "http://hackage.haskell.org/package/purescript-0.6.8/purescript-0.6.8.tar.gz"
+  sha1 "70fd4d3109d61c34c8898a30d222c4b1ad8fd7a5"
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc"
+
+  def install
+    install_cabal_package
+  end
+
+  test do
+    # TODO:  Figure out how to make use of the included cabal test suite.
+    # This is currently just compiling a near-trivial module, and
+    # checking to make sure the output file exists.
+    test_module_path = testpath/"Test.purs"
+    test_target_path = testpath/"test-module.js"
+    test_module_path.write <<-EOS.undent
+      module Test where
+      import Control.Monad.Eff
+      main :: forall e. Eff e Unit
+      main = return unit
+    EOS
+    system bin/"psc", test_module_path, "-o", test_target_path
+    unless File.exist?(test_target_path)
```

Comment:
I think running it would require node.js to be installed, unless you know a way of invoking the browser from the command line headlessly.